### PR TITLE
sundials: fix download URL and hash

### PR DESCRIPTION
**Pull request by wizeman**
###### Motivation for this change

The old URL doesn't seem to be available anymore (fails with 403 Forbidden error).

The tarball in the new URL contains a few changes from the old one, so the hash has changed.

Specifically, only the following files have changed:

* sundials-5.7.0/doc/arkode/ark_examples.pdf
* sundials-5.7.0/doc/arkode/ark_guide.pdf
* sundials-5.7.0/doc/cvode/cv_examples.pdf
* sundials-5.7.0/doc/cvode/cv_guide.pdf
* sundials-5.7.0/doc/cvodes/cvs_examples.pdf
* sundials-5.7.0/doc/cvodes/cvs_guide.pdf
* sundials-5.7.0/doc/ida/ida_examples.pdf
* sundials-5.7.0/doc/ida/ida_guide.pdf
* sundials-5.7.0/doc/idas/idas_examples.pdf
* sundials-5.7.0/doc/idas/idas_guide.pdf
* sundials-5.7.0/doc/kinsol/kin_examples.pdf
* sundials-5.7.0/doc/kinsol/kin_guide.pdf

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [X] Added a release notes entry if the change is major or breaking
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by wizeman** [comment]
This should be backported to 21.05.

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 125316` at 4ab604c4 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>78 packages skipped due to time constraints:</summary>
  <ul>
    <li>gama</li>
    <li>librsb</li>
    <li>minc_widgets</li>
    <li>octave</li>
    <li>octave-jit</li>
    <li>octaveFull</li>
    <li>octavePackages.arduino</li>
    <li>octavePackages.audio</li>
    <li>octavePackages.bim</li>
    <li>octavePackages.bsltl</li>
    <li>...</li>
  </ul>
</details>
<details>
  <summary>3 packages built successfully:</summary>
  <ul>
    <li>python38Packages.scikits-odes</li>
    <li>python39Packages.scikits-odes</li>
    <li>sundials</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
Also the download page links to github https://github.com/LLNL/sundials/releases/download/v5.7.0/cvode-5.7.0.tar.gz

File: pkgs/development/libraries/sundials/default.nix [hunk 1]
```diff
@@ -17,8 +17,8 @@ stdenv.mkDerivation rec {
   outputs = [ "out" "examples" ];
 
   src = fetchurl {
-    url = "https://computation.llnl.gov/projects/${pname}/download/${pname}-${version}.tar.gz";
-    sha256 = "jW3QlP7Mu41uzEE0DsFqZfq6yC7UQVAj9tfBwjkOovM=";
+    url = "https://github.com/LLNL/${pname}/releases/download/v${version}/${pname}-${version}.tar.gz";
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/sundials/default.nix:20]:
```suggestion
    url = "https://github.com/LLNL/sundials/releases/download/v${version}/sundials-${version}.tar.gz";
```

**Comment by wizeman** [comment]
Perhaps `fetchFromGitHub` is more appropriate?

**Comment by SuperSandro2000** [comment]
> Perhaps `fetchFromGitHub` is more appropriate?

We can't use that because we download a release asset.

**Pull request merged**